src/pydocstyle/checker.py

```diff
@@ -390,6 +411,174 @@ def check_starts_with_this(self, function, docstring):
+                                      len(dash_line.strip()))
+
+            line_after_dashes = \
+                context.following_lines[next_non_empty_line_offset + 1]
+            if line_after_dashes.strip() != '':
+                yield violations.D410(section_name)
+
+            if leading_space(dash_line) > indentation:
+                yield violations.D215(section_name)
+
+    @classmethod
+    def _check_section(cls, docstring, definition, context):
+        """D4{05,06,11}, D214: Section name checks.
+
+        Check for valid section names. Checks that:
+            * The section name is properly capitalized (D405).
+            * The section is not over-indented (D214).
+            * The section name has no superfluous suffix to it (D406).
+            * There's a blank line before the section (D411).
+
+        Also yields all the errors from `_check_section_underline`.
+        """
+        capitalized_section = context.section_name.title()
+        indentation = cls._get_docstring_indent(definition, docstring)
+
+        if (context.section_name not in cls.SECTION_NAMES and
+                capitalized_section in cls.SECTION_NAMES):
+            yield violations.D405(capitalized_section, context.section_name)
+
+        if leading_space(context.line) > indentation:
+            yield violations.D214(capitalized_section)
+
+        suffix = context.line.strip().lstrip(context.section_name)
+        if suffix != '':
+            yield violations.D406(capitalized_section, context.line.strip())
+
+        if context.previous_line.strip() != '':
+            yield violations.D411(capitalized_section)
+
+        for err in cls._check_section_underline(capitalized_section,
+                                                context,
+                                                indentation):
+            yield err
+
+    @check_for(Definition)
+    def check_docstring_sections(self, definition, docstring):
+        """D21{4,5}, D4{05,06,07,08,09,10}: Docstring sections checks.
+
+        Check the general format of a sectioned docstring:
+            '''This is my one-liner.
+
+            Short Summary
+            -------------
+
+            This is my summary.
+
+            Returns
+            -------
+
+            None.
+            '''
+
+        Section names appear in `SECTION_NAMES`.
+        """
+        if not docstring:
+            return
+
+        lines = docstring.split("\n")
+        if len(lines) < 2:
+            return
+
+        lower_section_names = [s.lower() for s in self.SECTION_NAMES]
+
+        def _suspected_as_section(_line):
+            result = self._get_leading_words(_line.lower())
+            return result in lower_section_names
+
+        # Finding our suspects.
+        suspected_section_indices = [i for i, line in enumerate(lines) if
+                                     _suspected_as_section(line)]
+
+        context = namedtuple('SectionContext', ('section_name',
```

Comment:
When using `namedtuple`, the assigned name should be the name of the type, e.g.:

```
SectionContext = namedtuple('SectionContext', ...)
```